appengine/flexible/websockets/main.py

```diff
@@ -0,0 +1,53 @@
+# Copyright 2017 Google Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import print_function
+
+# [START app]
+from flask import Flask, render_template
+from flask_sockets import Sockets
+
+
+app = Flask(__name__)
+sockets = Sockets(app)
+
+
+@sockets.route('/chat')
+def chat_socket(ws):
+    while not ws.closed:
+        message = ws.receive()
+        if message is None:  # message is "None" if the client has closed.
```

Comment:
Nit: Even with line 28's check, I take it this is necessary? (i.e. I presume `ws.closed` won't be true until the last `ws.receive()` call is made.)

Reply:
Yes, this is definitely necessary due to nuances on how receive() works in practice.